modules/nlu/src/backend/storage.ts

```diff
@@ -283,6 +287,22 @@ export default class Storage {
     }
   }
 
+  private _makeE2Fname(modelHash: string, lang: string): string {
+    return `${modelHash}.${lang}.model`
+  }
+
+  async writeE2Model(model: E2Model, modelHash: string): Promise<void> {
+    const strModel = serializeModel(model)
+    return this.botGhost.upsertFile(this.modelsDir, this._makeE2Fname(modelHash, model.languageCode), strModel)
+  }
+
+  async readE2Model(modelHash: string, languageCode: string): Promise<E2Model | undefined> {
+    if (await this.botGhost.fileExists(this.modelsDir, this._makeE2Fname(modelHash, languageCode))) {
```

Comment:
can re-use modelExists below?

Reply:
we can't. I figured that would be quite useless to re implement as the code is pretty simple but yes I can do.